### PR TITLE
Adding support for multiple state stores in the same cluster

### LINKED ISSUE #636: Support for multiple state stores on the same cluster
It will be great if we can deploy many instances of dapr and components on the same cluster.
```
annotations:
        dapr.io/dapr_id: "staging"
        dapr.io/enabled: "true"
        dapr.io/id: "bindings-pythonapp"
```
```
annotations:
        dapr.io/dapr_id: "production"
        dapr.io/enabled: "true"
        dapr.io/id: "bindings-pythonapp"
```

### PR DESCRIPTION
**Pull request by shalabhms**
# Description

Currently Dapr supports only one state store , this feature is to support multiple state stores in the same cluster

With this change , the state API allows the app to target a state store, for example:

v1.0/state/storeA/
v1.0/state/storeB/

This enables a scenario wherein an app can write to multiple stores based on need, one for fast in memory cache and another for persistent sql storage.

## Issue reference
Closes #636 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Unit tests passing
* [x] End-to-end tests passing
* [ ] Extended the documentation
* [ ] Specification has been updated
* [ ] Provided sample for the feature

### CODE REVIEW

File: pkg/runtime/runtime.go [hunk 1]
```diff
@@ -54,6 +54,7 @@ const (
 	appConfigEndpoint   = "dapr/config"
 	hostIPEnvVar        = "HOST_IP"
 	parallelConcurrency = "parallel"
+	defaultStore        = "defaultStore"
```
Comment (yaron2) [inline pkg/runtime/runtime.go]: Can you change to `defaultStateStore`?

File: pkg/runtime/runtime.go [hunk 2]
```diff
@@ -790,7 +803,8 @@ func (a *DaprRuntime) publishMessageGRPC(msg *pubsub.NewMessage) error {
 func (a *DaprRuntime) initActors() error {
 	actorConfig := actors.NewConfig(a.hostAddress, a.runtimeConfig.ID, a.runtimeConfig.PlacementServiceAddress, a.appConfig.Entities,
 		a.runtimeConfig.GRPCPort, a.appConfig.ActorScanInterval, a.appConfig.ActorIdleTimeout, a.appConfig.DrainOngoingCallTimeout, a.appConfig.DrainRebalancedActors)
-	act := actors.NewActors(a.stateStore, a.appChannel, a.grpc.GetGRPCConnection, actorConfig)
+	// TODO: provide a way to specify the actor state store, currently picking the first configured/default store
```
Comment (yaron2) [inline pkg/runtime/runtime.go]: This needs to be addressed.

We discussed reading a metadata property from the component spec which is `actorStore: true` to indicate this is the state store to be used for actors. If multiple stores exist, we pick the first one as default.

**Comment by yaron2** [comment]
> End-to-end tests failed.

@shalabhms e2e tests failed: run them locally to see why.

File: pkg/runtime/runtime.go (hunk 1 above)
Comment (shalabhms) [reply pkg/runtime/runtime.go]: done

File: pkg/runtime/runtime.go (hunk 2 above)
Comment (shalabhms) [reply pkg/runtime/runtime.go]: thanks @yaron2  for reviewing the code. I have pushed the suggested changes.

**Comment by shalabhms** [comment]
> > End-to-end tests failed.
> 
> @shalabhms e2e tests failed: run them locally to see why.

sure - checking locally on that .

**Comment by shalabhms** [comment]
> > End-to-end tests failed.
> 
> @shalabhms e2e tests failed: run them locally to see why.

@yaron2 , fixed the issue in state app, E2E tests passing now !

**Comment by yaron2** [comment]
Congrats indeed.

**Comment by yaron2** [comment]
Tests failing

**Comment by shalabhms** [comment]
> Tests failing
looking into it...

**Comment by yaron2** [comment]
There are conflicts...

**Comment by shalabhms** [comment]
> There are conflicts...

yes, I resolved it now.

**Comment by yaron2** [comment]
Should probably add `actorStateStore` attribute to all component YAMLs, not just for actors.

**Comment by yaron2** [comment]
Congrats

**Comment by shalabhms** [comment]
> Congrats

Thanks @yaron2  for review and guidance for this change !!!

**Pull request merged**